### PR TITLE
Update gjs to 1.65.2.

### PR DESCRIPTION
**Pull request by not-chicken**
Needed for #21382 

Compilation tested: 

- x86_64
- i686
- armv6l-musl
- armv7l
- aarch64

Runtime: 
- x86_64 - gnome-shell 3.36.2 seems to run fine.

Edit: I underestimated travis I guess. It's able to build this fine.

Patched up to disable spidermonkey checks for now. 
https://src.fedoraproject.org/rpms/mozjs68/blob/master/f/mozjs68.spec#_51

CC @Gottox

### CODE REVIEW

**Comment by sgn** [comment]
Please carry https://github.com/void-linux/void-packages/blob/master/srcpkgs/mozjs60/patches/0012-mozjs60.pc-honor-sysroot.patch over

**Comment by not-chicken** [comment]
> Please carry https://github.com/void-linux/void-packages/blob/master/srcpkgs/mozjs60/patches/0012-mozjs60.pc-honor-sysroot.patch over

Added, thanks!

**Comment by toluschr** [comment]
Compiles ans works fine!

Cloned using `git pull origin pull/21382/head pull/20413/head pull/21716/head`
Resolved merge conflicts (`srcpkgs/mutter/template`)
```
version=3.36.2
revision=1
```
and removed ` srcpkgs/mutter/patches/pipewire-0.3.patch`

**Comment by not-chicken** [comment]
- Updated mozjs to 68.8.0
- Added patch from openSUSE to remove unnecessary build dependencies.

**Comment by sgn** [comment]
I tested with mozjs68-68.7.0

Works fine with current GNOME.

File: common/shlibs [hunk 1]
```diff
@@ -830,9 +830,7 @@ libtotem.so.0 libtotem-3.6.2_1
 libclutter-1.0.so.0 clutter-1.14.0_1
 libavfs.so.0 libavfs-1.0.5_1
 libdmr.so.0.1 deepin-movie-reborn-3.2.5_1
-libgjs.so.0 gjs-1.63.3_1
-libgjs-gi.so gjs-1.63.3_1
-libgjs-dbus.so.0 gjs-1.63.3_1
```
Comment (not-chicken) [inline common/shlibs:835]: @sgn  Removed as they are no longer provided by the package.

**Comment by not-chicken** [comment]
> If noone steps in and complains, please ping me for a merge in end of Sunday ;)

Latest push passed the CI as well. I think it's ready to go!

File: common/shlibs (hunk 1 above)
Comment (sgn) [reply common/shlibs:835]: Is there any packages requiring them, right now?

Comment (sgn) [reply common/shlibs:835]: Look like no.

Comment (not-chicken) [reply common/shlibs:835]: Yep, nothing depends on them. I think `gjs` stopped providing them quite a while ago.

File: srcpkgs/mozjs68/template [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+		${_args}
+}
+
+do_check() {
+	dist/bin/jsapi-tests --format=none --exclude-random basic
+}
+
+post_install() {
+	# Remove unneeded static library
+	rm -f "${DESTDIR}"/usr/lib/*.ajs
```
Comment (sgn) [inline srcpkgs/mozjs68/template:59]: I'll move it to `-devel` instead.

Comment (not-chicken) [reply srcpkgs/mozjs68/template:59]: I removed it due to this https://gitlab.gnome.org/GNOME/jhbuild/issues/55
mozjs60 would probably still be needed for `cjs` ( it depends on mozjs52 atm)

Comment (not-chicken) [reply srcpkgs/mozjs68/template:59]: Turns out even `mozjs52-devel` installs the library with same name.

**Pull request closed without merging**